app/core/schools/endpoints_schools.py

```diff
@@ -0,0 +1,215 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.core import models_core, schemas_core
+from app.core.groups.groups_type import AccountType, GroupType
+from app.core.schools import cruds_schools
+from app.core.schools.schools_type import SchoolType
+from app.core.users import cruds_users
+from app.dependencies import (
+    get_db,
+    is_user_an_ecl_member,
+    is_user_in,
+)
+
+router = APIRouter(tags=["Schools"])
+
+
+@router.get(
+    "/schools/",
+    response_model=list[schemas_core.CoreSchoolSimple],
+    status_code=200,
+)
+async def read_schools(
+    db: AsyncSession = Depends(get_db),
+    user=Depends(is_user_an_ecl_member),
+):
+    """
+    Return all schools from database as a list of dictionaries
+    """
+
+    schools = await cruds_schools.get_schools(db)
+    return schools
+
+
+@router.get(
+    "/schools/{school_id}",
+    response_model=schemas_core.CoreSchool,
+    status_code=200,
+)
+async def read_school(
+    school_id: str,
+    db: AsyncSession = Depends(get_db),
+    user=Depends(is_user_in(GroupType.admin)),
+):
+    """
+    Return school with id from database as a dictionary. This includes a list of users being members of the school.
+
+    **This endpoint is only usable by administrators**
+    """
+
+    db_school = await cruds_schools.get_school_by_id(db=db, school_id=school_id)
+    if db_school is None:
+        raise HTTPException(status_code=404, detail="School not found")
+    return db_school
+
+
+@router.post(
+    "/schools/",
+    response_model=schemas_core.CoreSchoolSimple,
+    status_code=201,
+)
+async def create_school(
+    school: schemas_core.CoreSchoolBase,
+    db: AsyncSession = Depends(get_db),
+    user: schemas_core.CoreUser = Depends(is_user_in(GroupType.admin)),
+):
+    """
+    Create a new school.
+
+    **This endpoint is only usable by administrators**
+    """
+    if (  # We can't have two schools with the same name
+        await cruds_schools.get_school_by_name(school_name=school.name, db=db)
+        is not None
+    ):
+        raise HTTPException(
+            status_code=400,
+            detail=f"A school with the name {school.name} already exist",
+        )
+
+    try:
+        db_school = models_core.CoreSchool(
+            id=str(uuid.uuid4()),
+            name=school.name,
+            email_regex=school.email_regex,
+        )
+        await cruds_schools.create_school(school=db_school, db=db)
+        users = await cruds_users.get_users(
+            db=db,
+            included_account_types=[AccountType.external],
+        )
+        for db_user in users:
+            if re.match(db_school.email_regex, db_user.email):
+                await cruds_users.update_user(
+                    db,
+                    db_user.id,
+                    schemas_core.CoreUserUpdateAdmin(
+                        school_id=db_school.id,
+                        account_type=AccountType.other_school_student,
+                    ),
+                )
```

Comment:
This is clever, be I think it would be great to have a similar comportment on update. This is strange to modify users on school creation but on school modification.
Doing so is also a bit dangerous. If an admin makes a mistake when setting the regex, they may not be able to revert the induced changes.